Sources/libhostmgr/VM/VMBundle.swift

```diff
@@ -0,0 +1,166 @@
+import Foundation
+@preconcurrency import Virtualization
+import TSCBasic
+
+@available(macOS 13.0, *)
+public struct VMBundle: Sendable {
+    
+    internal struct BundlePathResolver {
+        let path: URL
+        
+        var configurationFilePath: URL {
+            self.path.appendingPathComponent("config.json")
+        }
+
+        var auxiliaryFileStorage: URL {
+            self.path.appendingPathComponent("aux.img")
+        }
+
+        var diskImageFilePath: URL {
+            self.path.appendingPathComponent("image.img")
+        }
+    }
+
+    struct ConfigFile: Codable {
+        let hardwareModelData: Data
+        let machineIdentifierData: Data
+
+        var hardwareModel: VZMacHardwareModel {
+            VZMacHardwareModel(dataRepresentation: hardwareModelData)!
+        }
+
+        var machineIdentifier: VZMacMachineIdentifier {
+            VZMacMachineIdentifier(dataRepresentation: machineIdentifierData)!
+        }
+
+        func write(to url: URL) throws {
+            try JSONEncoder().encode(self).write(to: url, options: .atomic)
+        }
+
+        static func from(url: URL) throws -> ConfigFile? {
+            let data = try Data(contentsOf: url)
+            return try JSONDecoder().decode(ConfigFile.self, from: data)
+        }
+    }
+
+    private let root: URL
+    private let hardwareModel: VZMacHardwareModel
+    private let machineIdentifier: VZMacMachineIdentifier
+
+    /// Persist the VM configuration to the local disk
+    func saveConfiguration() throws {
+        let pathResolver = BundlePathResolver(path: self.root)
+
+        try ConfigFile(
+            hardwareModelData: self.hardwareModel.dataRepresentation,
+            machineIdentifierData: self.machineIdentifier.dataRepresentation
+        )
+        .write(to: pathResolver.configurationFilePath)
+    }
+
+    var pathResolver: BundlePathResolver {
+        BundlePathResolver(path: self.root)
+    }
+}
+
+@available(macOS 13.0, *)
+extension VMBundle {
+    /// Instantiate a VMBundle from an existing VM package
+    ///
+    public static func fromExistingBundle(at url: URL) throws -> VMBundle {
+        let pathResolver = BundlePathResolver(path: url)
+
+        guard let configuration = try ConfigFile.from(url: pathResolver.configurationFilePath) else {
+            abort() // TODO :This should throw
+        }
+
+        return VMBundle(
+            root: url,
+            hardwareModel: configuration.hardwareModel,
+            machineIdentifier: configuration.machineIdentifier
+        )
+    }
+
+    /// Create a new VMBundle based on a restore image
+    ///
+    public static func createBundle(
+        named name: String,
+        fromRestoreImage image: VZMacOSRestoreImage,
+        withStorageCapacity capacity: Measurement<UnitInformationStorage> = .init(value: 64, unit: .gigabytes)
```

Comment:
64 GB isn't big enough for us. I guess we could either increase the default size here, or change the call site (or using a customizable CLI argument).

Reply:
I wasn't clear. The final VM image size is 43 GB. But 64 GB isn't big enough during setting up the VM, due to things like we need to install Xcode while the xip is still in the file system (or not removed at the most optimal timing).